### PR TITLE
Link login modal to follow a post

### LINKED ISSUE #4152: Link Login Modal to follow a post
We are creating a login/signup modal, next step is to link to various pages. In this issue we will be linking it to follow a post
link - https://publiclab.org/notes/joyofsoy/11-09-2018/fighting-the-flood-community-activism-and-education-in-pensacola
        `app/views/users/_follow.html.erb`

### PR DESCRIPTION
**Pull request by dinaelhanan**
Fixes #4152

* [x] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `rake test`
* [x] code is in uniquely-named feature branch and has no merge conflicts 📁
* [x] PR is descriptively titled 📑
* [ ] screenshots/GIFs are attached 📎 in case of UI updation
* [x] ask `@publiclab/reviewers` for help, in a comment below

### CODE REVIEW

**Comment by dinaelhanan** [comment]
@publiclab/reviewers

**Comment by oorjitchowdhary** [comment]
@sidharthbansal aren't #4437 and #4152 the same issues? 

@dinaelhanan I think you have made changes to a wrong file..
I don't think this is the file for the buttons @sidharthbansal was talking about

**Comment by oorjitchowdhary** [comment]
https://github.com/publiclab/plots2/blob/699108b54624cf0554c6bb87d4f05724f79901ad/app/views/like/_like.html.erb 
This is the file for the buttons

**Comment by dinaelhanan** [comment]
@oorjitchowdhary thanks!

**Comment by SidharthBansal** [comment]
@dinaelhanan please rebase and resolve conflicts so that @oorjitchowdhary or I or anyone else can help you further.
Also, please provide the gifs.
Thanks

**Comment by dinaelhanan** [comment]
@SidharthBansal I am trying on my localhost and it's not working, I click the button and nothing happens..Please check out my changes. Thanks

**Comment by dinaelhanan** [comment]
@publiclab/reviewers would appreciate some guidance on this, thanks!

**Comment by SidharthBansal** [comment]
My laptop crashed yesterday again. Now it is fine. Sorry was not able to
help you. I will help you after getting up from sleep.

**Comment by oorjitchowdhary** [comment]
I think we should wait on this until #4453 is merged. That may be the issue here..

**Comment by SidharthBansal** [comment]
Yeah we need to merge that before this PR.
Thanks Oorjit. Dina you can work on Amy other issue in the meantime.

**Comment by oorjitchowdhary** [comment]
#4453 is merged.. @dinaelhanan update your branch.. And try adding the 'requireLogin' class to the follow button.. It should work then

**Comment by dinaelhanan** [comment]
@oorjitchowdhary OK I've updated and added requireLogin class to the buttons. Please checkout my changes, I'm not quite sure if I've added the class correctly. Thanks for all the help :+1:

File: app/views/like/_like.html.erb [hunk 1]
```diff
@@ -10,6 +10,19 @@
...
+      <i class="fa fa-user-plus" aria-hidden="true"></i> 
     <li title="Follow" class="btn btn-default btn-sm" data-html="true" rel="popover" data-content="<%= "No tags" if tagnames.nil? || tagnames.length == 0 %><% if tagnames %><% tagnames.each do |tagname| %><p style='margin-bottom:3px;'><a href='/subscribe/tag/<%= tagname %>' class='btn btn-default btn-mini'><%= tagname %></a></p><% end %><% end %><hr /><i class='fa fa-user'></i><% if current_user && !current_user.following?(node.author) %>&nbsp;<a class='btn btn-sm' href='/relationships?followed_id=<%= node.author.id %>' data-method='post' > <%= node.author.name %></a><% else %><%= node.author.name %><% end %>" title="Follow by tag or author" data-placement="left">
```
Comment (oorjitchowdhary) [inline app/views/like/_like.html.erb]:
```diff
- <li title="Follow" class="btn btn-default btn-sm" data-html="true" rel="popover" data-content="<%= "No tags" if tagnames.nil? || tagnames.length == 0 %><% if tagnames %><% tagnames.each do |tagname| %><p style='margin-bottom:3px;'><a href='/subscribe/tag/<%= tagname %>' class='btn btn-default btn-mini'><%= tagname %></a></p><% end %><% end %><hr /><i class='fa fa-user'></i><% if current_user && !current_user.following?(node.author) %>&nbsp;<a class='btn btn-sm' href='/relationships?followed_id=<%= node.author.id %>' data-method='post' > <%= node.author.name %></a><% else %><%= node.author.name %><% end %>" title="Follow by tag or author" data-placement="left">
+ <li title="Follow" class="btn btn-default btn-sm requireLogin" data-html="true" rel="popover" data-content="<%= "No tags" if tagnames.nil? || tagnames.length == 0 %><% if tagnames %><% tagnames.each do |tagname| %><p style='margin-bottom:3px;'><a href='/subscribe/tag/<%= tagname %>' class='btn btn-default btn-mini'><%= tagname %></a></p><% end %><% end %><hr /><i class='fa fa-user'></i><% if current_user && !current_user.following?(node.author) %>&nbsp;<a class='btn btn-sm' href='/relationships?followed_id=<%= node.author.id %>' data-method='post' > <%= node.author.name %></a><% else %><%= node.author.name %><% end %>" title="Follow by tag or author" data-placement="left">
```

Comment (oorjitchowdhary) [reply app/views/like/_like.html.erb]: Remove all the added lines and just add `requireLogin` in the class as shown in the diff above and in that line only.. And then try it out

This is only a line change..

Comment (dinaelhanan) [reply app/views/like/_like.html.erb]: @oorjitchowdhary Ok thanks for that, I've done it :+1:

**Comment by jywarren** [comment]
In the rails console (`rails console`) you can run `User.find(1).unban` --
sorry, there's an open issue where we're trying to debug this! #608 i
think? Bug recently cropped up!

**Comment by dinaelhanan** [comment]
@jywarren ok thanks for that! I've done it and it works. so this is what I'm getting. I think that's what I was supposed to do:

followflow

**Comment by dinaelhanan** [comment]
@SidharthBansal I did click on the follow button, that little "blog" button dialog pops up after you click follow on the local version and the actual site (this gif is from the actual site. On the previous gif, you can see I did the same and clicked on what part of the note i would like to follow):
screencast20190106140438

**Comment by SidharthBansal** [comment]
Checking this on my laptop.

**Comment by SidharthBansal** [comment]
Pushed these changes on the unstable branch

**Comment by SidharthBansal** [comment]
It is working.
Thanks Dina. Merging it up @jywarren.

**Pull request merged**